packages/compile-common/src/compilerSupplier/supplier.ts

```diff
@@ -0,0 +1,139 @@
+import type { Strategy as FreeStrategy, BaseStrategy } from "./strategy";
+import type { Results as FreeResults } from "./results";
+
+export namespace Supplier {
+  /**
+   * Type-level description of a supplier
+   */
+  export type Specification = {
+    options: any;
+    results: FreeResults.Specification;
+    strategies: {
+      [strategyName: string]: Omit<FreeStrategy.Specification, "results">;
```

Comment:
This should probably be changed so that the strategies' results types have to agree with that of the supplier, rather than being omitted.

Reply:
done!